gtest_parallel_tests.py

```diff
@@ -446,5 +446,207 @@ def root():
+      test_group, test = test_name.split('.')
+      return self._test_data.get(test_binary, {}).get(test_group, {}).get(test, None)
+
+  class MockPopen(object):
+    class MockSubprocess(object):
+      def __init__(self, stdout=""):
+        self.stdout = stdout
+
+      def communicate(self):
+        return [self.stdout]
+
+    def __init__(self, test_data=None):
+      self._test_data = test_data
+      self.last_invocation = None
+
+    def __call__(self, command, stdout=None):
+      self.last_invocation = command
+      binary = command[0]
+      test_list = []
+      tests_for_binary = sorted(self._test_data.get(binary, {}).iteritems())
+      for test_group, tests in tests_for_binary:
+        test_list.append(test_group + ".")
+        for test in sorted(tests):
+          test_list.append("  " + test)
+      return self.MockSubprocess('\n'.join(test_list))
+
+  def test_tasks_are_sorted(self):
+    fake_tests = {
+        "fake_unittests": {
+            "FakeTest": {
+                "TestSomething": None,
+                "TestSomethingElse": 2,
+            },
+            "SomeOtherTest": {
+                "YetAnotherTest": 3,
+            },
+        },
+        "fake_tests": {
+            "Foo": {
+                "Bar": 4,
+                "Baz": 4,
+            }
+        }
+    }
+
+    with guard_patch_module('subprocess.Popen', self.MockPopen(fake_tests)):
+      tasks = gtest_parallel.find_tests(fake_tests.keys(), [],
+                                        self.MockOptions(),
+                                        self.MockTestTimes(fake_tests))
+    self.assertEqual(tasks, sorted(tasks))
+
+  def test_runs_disabled_tests_when_asked(self):
+    fake_tests = {
+        "fake_unittests": {
+            "FakeTest": {
+                "Test1": None,
+                "DISABLED_Test2": None,
+            }
+        }
+    }
+    mock_Popen = self.MockPopen(fake_tests)
+
+    with guard_patch_module('subprocess.Popen', mock_Popen):
+      tasks = gtest_parallel.find_tests(
+          fake_tests.keys(), [],
+          self.MockOptions(gtest_also_run_disabled_tests=False),
+          self.MockTestTimes(fake_tests)
+      )
+    self.assertEqual(len(tasks), 1)
+    self.assertFalse("DISABLED_" in tasks[0].test_name)
+    self.assertNotIn("--gtest_also_run_disabled_tests", mock_Popen.last_invocation)
+
+    with guard_patch_module('subprocess.Popen', mock_Popen):
+      tasks = gtest_parallel.find_tests(
+          fake_tests.keys(), [],
+          self.MockOptions(gtest_also_run_disabled_tests=True),
+          self.MockTestTimes(fake_tests)
+      )
+    self.assertEqual(len(tasks), 2)
+    self.assertEqual(sorted([task.test_name for task in tasks]),
+                     ["FakeTest.DISABLED_Test2", "FakeTest.Test1"])
+    self.assertIn("--gtest_also_run_disabled_tests", mock_Popen.last_invocation)
+
+  def test_runs_only_failed_tests_when_asked(self):
+    fake_tests = {
```

Comment:
Can these be shared as data, kOneFailedTestOnePassing or something. But proper style.